perf(event attachment): Add file to index

nathan-heskia opened the pull request:
Since `file` isn't currently indexed with the project_id and date, queries that access the file property will fetch the event attachment rows from disk in order to retrieve the file_id. This is causing significant i/o for larger queries so adding `file` to the index will reduce the query time for these cases.

Edit: only adding in the index in this pr, will remove the current in a separate one.

SQL query:
```
--
-- Custom state/database change combination
--

CREATE INDEX CONCURRENTLY "sentry_eventattachment_project_id_date_added_fi_f3b0597f_idx" ON "sentry_eventattachment" ("project_id", "date_added", "file_id");
 ;
```

wedamija commented on src/sentry/migrations/0060_eventattachment_index.py:
```diff
@@ -0,0 +1,50 @@
+                        """,
+                )
+            ],
+            state_operations=[
+                migrations.AlterIndexTogether(
+                    name='eventattachment',
+                    index_together=set([('project_id', 'date_added', 'file')]),
+                ),
+            ],
```
Just noting that all this will do is update Django's migration state. So as far as it is concerned, we have created the new index and dropped the old. In reality, after this runs we will have only created the new index and the old one will still be in place. Or possibly, even the new index might not be created, since we'll likely have some lag between this being deployed and this being run manually.

nathan-heskia commented:
Thanks for your feedback @mattrobenolt and @wedamija. I've modified this PR to add another index concurrently and moved `AlterIndex` into `StateOperations`. In a separate PR I'll drop the current index.

mattrobenolt commented on src/sentry/models/eventattachment.py:
```diff
@@ -32,7 +32,7 @@ class EventAttachment(Model):
         app_label = "sentry"
         db_table = "sentry_eventattachment"
-        index_together = (("project_id", "date_added"),)
+        index_together = (("project_id", "date_added", "file"),)
```
This doe snot reflect what is happening here. This should just suggest that we're adding another index.

```
index_together = (
    ("project_id", "date_added",),
    ("project_id", "date_added", "file_id",),
)
```

is correct based on what we're doing. The migration is simply adding a new one. I think this then goes back to @wedamija's comment then about the state operation not matching.

wedamija replied:
I actually suggested that he handle the migration in this way, just in terms of not having to be overly complex when changing the state. But if you prefer we could be more explicit and remove the old index in the followup pr.

mattrobenolt replied:
I'd prefer we do it correctly in two steps so the state reflects reality. It won't really hurt having the extra index, even if for example we never drop the other.

nathan-heskia replied:
@mattrobenolt I added the additional index. I'll create a follow up PR which removes the first index without the file.
